### ISSUE TITLE
accelerator.gather() for non-tensor inputs?

### ISSUE THREAD

**Issue by ttcqaq** [huggingface/accelerate#86]
Can I use gather() to merge two non-tensor inputs? For example:
One process has a list of string like [ ['a', 'aa'], ['b', 'bb'] ], another has a list like [ ['c', 'cc'], ['d', 'dd'] ].
Is there any way to merge these two lists into [ ['a', 'aa'], ['b', 'bb'], ['c', 'cc'], ['d', 'dd'] ]?
I think it's necessary when evaluating the model, only gathering the logits and labels may not be enough sometimes.

**Comment by sgugger** [comment]
There is no way to gather strings inputs without going through integers and a tensor, AFAIK. What is your use case? We have distributed evaluation working for [summarization](https://github.com/huggingface/transformers/blob/master/examples/pytorch/summarization/run_summarization_no_trainer.py) and [translation](https://github.com/huggingface/transformers/blob/master/examples/pytorch/translation/run_translation_no_trainer.py) in Transformers using Accelerate.

**Comment by ttcqaq** [comment]
Thanks for your reply! My case is like that:
``` python
for inputs, targets, xxxx in validation_dataloader:
    predictions = model(inputs)
    # Gather all predictions and targets
    all_predictions = accelerator.gather(predictions)
    all_targets = accelerator.gather(targets)
    # Example of use with a `Datasets.Metric`
    custom_metric.add_batch(all_predictions, all_targets, xxxx)
```
`xxxx` is another non-tensor input (list or nested list) like [ ['a', 'aa'], ['b', 'bb'] ].
accelerator.gather() can help the tensors `inputs` and `targets` (with size `batch_size`) merge from all processes, and yields the tensors `all_predictions` and `all_targets` with sizes of `batch_size * num_processes`, but after the line of `accelerator.gather()`, non-tensor variable `xxxx` remains the length of `batch_size`, which may cause some difficulty to the following metric calculating process.

My final solution is just converting the `xxxx` to tensor according to some vocabs, and it works.

**Comment by TingchenFu** [comment]
Hi, @antct  do you find a way to gather non-tensor variables? could you share your solution? Thanks!
